pkg/storaged/dialogx.jsx

```diff
@@ -462,3 +464,128 @@ export const Skip = (className, options) => {
         }
     };
 };
+
+const StatelessSlider = ({ fraction, onChange }) => {
+    function start_dragging(event) {
+        let el = event.currentTarget;
+        let width = el.offsetWidth;
+        let left = el.offsetLeft;
+        while (el.offsetParent) {
+            el = el.offsetParent;
+            left += el.offsetLeft;
+        }
+
+        function drag(event) {
+            let f = (event.pageX - left) / width;
+            if (f < 0) f = 0;
+            if (f > 1) f = 1;
+            onChange(f);
+        }
+
+        function stop_dragging() {
+            document.removeEventListener("mousemove", drag);
+            document.removeEventListener("mouseup", stop_dragging);
+        }
+
+        document.addEventListener("mousemove", drag);
+        document.addEventListener("mouseup", stop_dragging);
+        drag(event);
+    }
+
+    if (fraction < 0) fraction = 0;
+    if (fraction > 1) fraction = 1;
+
+    return (
+        <div className="slider" onMouseDown={start_dragging}>
+            <div className="slider-bar" style={{ width: fraction * 100 + "%" }}>
+                <div className="slider-thumb" />
+            </div>
+        </div>
+    );
+};
+
+class SizeSliderElement extends React.Component {
+    constructor(props) {
+        super();
+        this.units = cockpit.get_byte_units(props.value || props.max);
+        this.units.forEach(u => { if (u.selected) this.state = { unit: u.factor }; });
+    }
+
+    render() {
+        let { val, max, onChange } = this.props;
+        let { unit } = this.state;
+
+        const change_slider = (f) => {
+            onChange(Math.round(f * max));
+        };
+
+        const change_text = (event) => {
+            if (event.type == "change") {
+                let val = +event.target.value * unit;
+                if (event.target.value === "" || isNaN(val)) {
+                    /* If there something else than a number in the
+                       input element, we use that as the value
+                       directly so that it sticks around.  It will be
+                       rejected by the validate function below.
+                     */
+                    onChange(event.target.value);
+                } else {
+                    onChange(Math.round(val));
+                }
+            }
+        };
+
+        const change_unit = (u) => {
+            this.setState({ unit: +u });
```

Comment:
Same lone `+` here, so I begin to think this is on purpose, and I just don't understand it?